app/src/main/java/de/danoeh/antennapod/fragment/ExternalPlayerFragment.java

```diff
@@ -127,8 +130,8 @@ public void onPlaybackEnd() {
     @Override
     public void onResume() {
         super.onResume();
-        controller.init();
         onPositionObserverUpdate();
+        new Thread(() -> controller.init()).start();
```

Comment:
Do you actually do any heavy lifting here that needs to happen in the background?
If so, why do we still do `controller.init()` in `connectToPlaybackService()` which definitely is executed on the UI thread?